vertx-web/src/main/java/io/vertx/ext/web/Router.java

```diff
@@ -37,6 +39,44 @@
 @VertxGen
 public interface Router extends Handler<HttpServerRequest> {
 
+  /**
+   * add metadata to this router. Used for saved extra data.
+   *
+   * @param key the metadata of key
+   * @param value the metadata of value
+   * @return a reference to this, so the API can be used fluently
+   */
+  @Fluent
+  default Router metadata(String key, Object value) {
```

Comment:
Same comment as previously, I'd prefer a builder like pattern instead for consistency